test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs

```diff
@@ -254,5 +258,74 @@ public void StaticPipeAssertKeys()
+            var dataPath = GetDataPath("external", "winequality-white.csv");
+            var dataSource = new MultiFileSource(dataPath);
+
+            var reader = TextLoader.CreateReader(env,
+                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+                separator: ';', hasHeader: true);
+            var data = reader.Read(dataSource);
+
+            var est = reader.MakeNewEstimator()
+                .Append(r => (r.label, r.features, bin: r.features.NormalizeByBinning(), mm: r.features.NormalizeByMinMax()));
+            var tdata = est.Fit(data).Transform(data);
+
+            var schema = tdata.AsDynamic.Schema;
+            Assert.True(schema.TryGetColumnIndex("features", out int featCol));
+            Assert.True(schema.TryGetColumnIndex("bin", out int binCol));
+            Assert.True(schema.TryGetColumnIndex("mm", out int mmCol));
+            Assert.False(schema.IsNormalized(featCol));
+            Assert.True(schema.IsNormalized(binCol));
+            Assert.True(schema.IsNormalized(mmCol));
+        }
+
+        [Fact]
+        public void NormalizerWithOnFit()
+        {
+            var env = new TlcEnvironment(seed: 0);
+            var dataPath = GetDataPath("external", "winequality-white.csv");
+            var dataSource = new MultiFileSource(dataPath);
+
+            var reader = TextLoader.CreateReader(env,
+                c => c.LoadFloat(0, 2),
+                separator: ';', hasHeader: true);
+            var data = reader.Read(dataSource);
+
+            // These will be populated once we call fit.
+            ImmutableArray<float> mm;
+            ImmutableArray<float> ss;
+            ImmutableArray<ImmutableArray<float>> bb;
+
+            var est = reader.MakeNewEstimator()
+                .Append(r => (r, 
+                    ncdf: r.NormalizeByCumulativeDistribution(onFit: (m, s) => mm = m),
+                    n: r.NormalizeByMeanVar(onFit: (s, o) => { ss = s; Assert.Empty(o); }),
+                    b: r.NormalizeByBinning(onFit: b => bb = b)));
+            var tdata = est.Fit(data).Transform(data);
+
+            Assert.Equal(3, mm.Length);
+            Assert.Equal(3, ss.Length);
+            Assert.Equal(3, bb.Length);
+
+            // Just for fun, let's also write out some of the lines of the data to the console.
+            using (var stream = new MemoryStream())
+            {
+                IDataView v = new ChooseColumnsTransform(env, tdata.AsDynamic, "r", "ncdf", "n", "b");
+                v = TakeFilter.Create(env, v, 10);
+                var saver = new TextSaver(env, new TextSaver.Arguments()
+                {
+                    Dense = true,
+                    Separator = ",",
+                    OutputHeader = false
+                });
+                saver.SaveData(stream, v, Utils.GetIdentityPermutation(v.Schema.ColumnCount));
+                Console.WriteLine(Encoding.UTF8.GetString(stream.ToArray()));
```

Comment:
>Console.WriteLine(Encoding.UTF8.GetString(stream.ToArray())); [](start = 16, length = 61)

do you need it, or it was part of debugging code?

Reply:
Hi @Ivanidzo4ka . "Need" is a bit of a tricky word. I'd rather say my including it was deliberate, to be sure... indeed, I've had this test base class that captures console output for a few PRs, despite the fact that it (should) not affect the outcome of the tests. Nonetheless I do find this sort of output incredibly useful.